homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,301 @@
+        _LOGGER.debug('hold_time_service time: %s', time)
+
+        if entity_id:
+            target_thermostats = [device for device in devices
+                                  if device.entity_id in entity_id]
+        else:
+            target_thermostats = devices
+
+        for thermostat in target_thermostats:
+            thermostat.async_set_preset_period(time)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_HOLD_TIME, hold_time_service,
+        schema=HOLD_PERIOD_SCHEMA)
+
+
+async def async_unload_entry(
+        hass: HomeAssistantType, entry: ConfigType
+) -> bool:
+    """Unload Lyric thermostat config entry."""
+    hass.services.async_remove(DOMAIN, SERVICE_HOLD_TIME)
+
+
+class LyricThermostat(LyricDeviceEntity, ClimateDevice):
+    """Representation of a Lyric thermostat."""
+
+    def __init__(self, device, location, temp_unit, hass) -> None:
+        """Initialize the thermostat."""
+        unique_id = '{}_climate'.format(device.macID)
+        self._unit = temp_unit
+
+        # Not all lyric devices support cooling and heating remove unused
+        self._operation_list = [HVAC_MODE_OFF]
+
+        # Add supported lyric thermostat features
+        if device.can_heat:
+            self._operation_list.append(HVAC_MODE_HEAT)
+
+        if device.can_cool:
+            self._operation_list.append(HVAC_MODE_COOL)
+
+        if device.can_heat and device.can_cool:
+            self._operation_list.append(HVAC_MODE_HEAT_COOL)
+
+        # data attributes
+        self._location = None
+        self._name = None
+        self._humidity = None
+        self._target_temperature = None
+        self._setpoint_status = None
+        self._temperature = None
+        self._temperature_scale = None
+        self._target_temp_heat = None
+        self._target_temp_cool = None
+        self._dual_setpoint = None
+        self._mode = None
+        self._min_temperature = None
+        self._max_temperature = None
+        self._next_period_time = None
+        self._schedule_type = None
+        self._schedule_sub_type = None
+        self._current_schedule_period = None
+        self._current_schedule_period_day = None
+        self._vacation_hold = None
+
+        super().__init__(device, location, unique_id, None, None)
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        return self._temperature_scale
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._temperature
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac mode ie. heat, cool, off."""
+        hvac_mode = None
+        if self._mode in [HVAC_MODE_HEAT_COOL, HVAC_MODE_HEAT, HVAC_MODE_COOL,
```

Comment:
It's fragile to couple the lyric api directly to the home assistant api. If the home assistant api changes, the integration would break until the lyric library is updated.

I recommend making two dicts that map between the home assistant modes and the lyric device modes.